public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -98,15 +99,14 @@ export class PanelEditor extends SceneObjectBase<PanelEditorState> {
     const dashboard = this.state.dashboardRef.resolve();
     const sourcePanel = this.state.sourcePanelRef.resolve();
 
-    const panelMngr = this.state.panelRef.resolve();
-
     if (!dashboard.state.isEditing) {
       dashboard.onEnterEditMode();
     }
 
-    const newState = sceneUtils.cloneSceneObjectState(panelMngr.state.panel.state);
-
-    sourcePanel.setState(newState);
+    const panelMngr = this.state.panelRef.resolve();
+    if (sourcePanel.parent instanceof SceneGridItem) {
+      sourcePanel.parent.setState({ body: panelMngr.state.panel });
+    }
```

Comment:
This is problematic as the `parent` for the panel manager's `state.panel` is already set. This this will result in a warning in the console.

I wonder why the state clone does not work?

Reply:
> I wonder why the state clone does not work?

I think because VizPanel's _plugin property isn't actually part of the state

Reply:
Ah, you may be right!  But I think what we can do is actually is clone the managed panel when navigating back:

```
sourcePanel.parent.setState({ body: panelMngr.state.panel.clone() });
```

I think this should solve the problem, but may have memory implications...

Reply:
Otherwise we would have to implement the state change handler in the `VizPanel` directly in order to support VizPanel plugin type change :(

Reply:
Right, I think that was something that was mentioned as something to avoid in the initial panel edit PR

Reply:
Anyway, will add the clone change and see what happens